frontend/app/features/kanban/components/KanbanBoard.tsx

```diff
@@ -1,19 +1,61 @@
 "use client";
 
+import { useGetAllTasksQuery } from "@store/api/taskSlice";
+import { useState } from "react";
+import { useTaskFilters } from "../hooks/useTaskFilters";
 import styles from "../styles/kanban.module.css";
-import CreateTaskForm from "./CreateTaskForm";
+import TaskFilter from "./TaskFilter";
 
 const KanbanBoard: React.FC = () => {
+  const { register, handleSubmit, resetFilters, applyFilters, getFilteredQuery, uniqueCreators, uniqueAssignees } =
+    useTaskFilters();
+
+  const [filters, setFilters] = useState(getFilteredQuery());
```

Comment:
whats this for?, isn't this still a second state for filters???, like I recommended before, please use only useForm for this.


Technical debt:

use oly useForm for collecting data, and on submit, save the submitted filtes to url search  param, on the other hand, pass url search param to `useGetAllTasksQuery` (I see you are using useGetAllTasksQuery in 2 places already which is not good).
so the flow is, `useGetAllTasksQuery` fetches the data based on url search param, when user filters and clicks on submit, you just chagne to url search param. this approach also allows you to persist filters on reload or sharing filtered links between users